db/music_pg.sql

```diff
@@ -0,0 +1,373 @@
+);
+
+CREATE TABLE "artist_genre" (
+  "artist_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "artist_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "participant" (
+  "participant_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "username" text UNIQUE NOT NULL,
+  "nickname" text,
+  "date_joined" timestamp NOT NULL DEFAULT (now()),
+  "time_zone" text,
+  "active" bool_int NOT NULL
+);
+
+CREATE TABLE "participant_nickname" (
+  "participant_nickname_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "nickname" text NOT NULL,
+  "date_start" date NOT NULL,
+  "date_end" date
+);
+
+CREATE TABLE "genre" (
+  "genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text UNIQUE NOT NULL,
+  "parent_id" text
+);
+
+CREATE TABLE "song_genre" (
+  "song_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "song_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "bracket" (
+  "bracket_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "epoch" int,
+  "cycle" int,
+  "name" text,
+  "type" text NOT NULL,
+  "year_min" int,
+  "year_max" int,
+  "order" int NOT NULL,
+  "size" text NOT NULL,
+  "date_start" timestamp,
+  "date_end" timestamp,
+  "discord_message_id" text
+);
+
+CREATE TABLE "project" (
+  "project_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text UNIQUE,
+  "project_lead_id" int
+);
+
+CREATE TABLE "bracket_selection" (
+  "bracket_selection_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "project_id" int NOT NULL,
+  "epoch" int NOT NULL,
+  "cycle" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "date_selected" timestamp NOT NULL
+);
+
+CREATE TABLE "seed" (
+  "seed_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "bracket_id" int NOT NULL,
+  "seed" int NOT NULL,
+  "submission_id" int NOT NULL,
+  "emoji" text
+);
+
+CREATE TABLE "seed_history" (
+  "seed_history_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "date_changed" timestamp NOT NULL,
+  "reason_changed" text NOT NULL,
+  "seed_id" int NOT NULL,
+  "submission_id" int NOT NULL,
+  "emoji" text
+);
+
+CREATE TABLE "round" (
+  "round_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "bracket_id" int NOT NULL,
+  "name" text NOT NULL,
+  "size" int NOT NULL,
+  "index_asc" int NOT NULL,
+  "index_desc" int NOT NULL,
+  "date_start" timestamp NOT NULL,
+  "date_end" timestamp,
+  "discord_message_id" text
+);
+
+CREATE TABLE "match" (
+  "match_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "round_id" int NOT NULL,
+  "index" int NOT NULL,
+  "date_posted" timestamp NOT NULL,
+  "tie" text,
+  "winner_id" int,
+  "date_cancelled" timestamp,
+  "reason_cancelled" text,
+  "next_match_id" int,
+  "discord_message_id" text
+);
+
+CREATE TABLE "match_seed" (
+  "match_seed_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "match_id" int NOT NULL,
+  "seed_id" int NOT NULL,
+  "index" int NOT NULL,
+  "votes" int,
+  "voters" text[],
```

Comment:
I think we'll probably want this to be `NOT NULL` and just be an empty array until we have a voter for it. The only reason I could see for not using an empty array as the default null state is if we want to have this be `NULL` until a match is actually posted and live, and then update it to an empty array (that is, a null value would indicate that nobody has had the opportunity to vote on it and an empty array would indicate that we've had the opportunity but nobody has done so).

Reply:
Oh, sure. I hadn't realized an empty array counted as not null.

Reply:
I'm pretty sure it does, anyway. I guess we'll find out soon enough when we try to start adding data in haha